magento/pwa-studio
# [SPIKE] [PWA-339] Conditionally bundle CE or EE features in Venia.

## Description

Conditionally bundle CE or EE features in Venia.

This is a webpack plugin that at build time, swaps code depending on Magento backend edition. It uses `FILENAME.ce.js` in case of CE and `FILENAME.ee.js` in case of EE. This solution is scalable and maintainable even if we come up with different versions of the backend in the future.

As an example, I have added a couple examples of different components and different queries.

## Related Issue
Closes [PWA-339](https://jira.corp.magento.com/browse/PWA-339)

### Verification Stakeholders
@zetlen @jimbo 

### Specification
Docs need to be updated to reflect the new variable in .env `MAGENTO_BACKEND_EDITION`. @jcalcaben 

### Verification Steps
1. Run `yarn build`.
2. Inspect the build, EE code should not be present in the bundle, or in case of EE, the CE code should not be present.

## Screenshots / Screen Captures (if appropriate)

![image](https://user-images.githubusercontent.com/35203638/73274761-ba5b9a80-41ab-11ea-8b69-d60e396202dc.png)

![image](https://user-images.githubusercontent.com/35203638/73275818-4fab5e80-41ad-11ea-84d7-bf21241c02df.png)

## Checklist
* Docs need to be updated to reflect the new `MAGENTO_BACKEND_EDITION` environment variable.
